src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -0,0 +1,390 @@
+                }
+                i++;
+            }
+        }
+    }
+
+    /**
+     * Get the first task in the list.
+     * @return the building which needs a delivery.
+     */
+    @Nullable
+    public AbstractBuilding getTask()
+    {
+        if(list.isEmpty())
+        {
+            return null;
+        }
+        return list.remove(0);
+    }
+
+    /**
+     * Check if the required items by the building are in the wareHouse.
+     * @param buildingEntry the building requesting.
+     * @return true if has something in warehouse to deliver.
+     */
+    public boolean checkInWareHouse(final AbstractBuilding buildingEntry)
+    {
+        if(buildingEntry.areItemsNeeded())
+        {
+            for(ItemStack stack : buildingEntry.getNeededItems())
+            {
+                if(isInHut(stack))
+                {
+                    buildingEntry.setOnGoingDelivery(true);
+                    list.add(buildingEntry);
+                    return true;
+                }
+            }
+        }
+
+        String tool = buildingEntry.getRequiredTool();
+        if(!tool.isEmpty() && isToolInHut(tool))
+        {
+            buildingEntry.setOnGoingDelivery(true);
+            list.add(buildingEntry);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Check all chests in the worker hut for a required item.
+     * @param is the type of item requested (amount is ignored)
+     * @return true if a stack of that type was found
+     */
+    public boolean isInHut(@Nullable final ItemStack is)
+    {
+        @Nullable final AbstractBuilding building = getBuilding();
+
+        boolean hasItem;
+        if(building != null)
+        {
+            hasItem = isInTileEntity(building.getTileEntity(), is);
+
+            if(hasItem)
+            {
+                return true;
+            }
+
+            for(BlockPos pos : building.getAdditionalCountainers())
+            {
+                TileEntity entity = worldObj.getTileEntity(pos);
+                if(entity instanceof TileEntityChest)
+                {
+                    hasItem = isInTileEntity((TileEntityChest) entity, is);
+
+                    if(hasItem)
+                    {
+                        return true;
+                    }
+                }
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Check for a certain item and return the position of the chest containing it.
+     * @param stack the stack to search for.
+     * @return the position or null.
+     */
+    @Nullable
+    public BlockPos getPositionOfChestWithItemStack(ItemStack stack)
+    {
+        @Nullable final AbstractBuilding building = getBuilding();
+
+        if(building != null)
+        {
+            if(isInTileEntity(building.getTileEntity(), stack))
+            {
+                return building.getLocation();
+            }
+
+            for(BlockPos pos : building.getAdditionalCountainers())
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'pos' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)